OSU-CS361-W17/group20_project3
# proj2 code cleanup

Fixes #6 and #5 (all bullets except the last, which will probably be introduced in a separate PR once our new ``Ship`` model has settled down)

## Changes this PR introduces
- [x] deletes unused submodule
- [x] fixes backend code so that ships' ending squares correctly correspond to the ships' respective lengths
- [x] fix tests accordingly
- [x] update offboard/overlap error checking functionality accordingly
- [x] prevent error popups from triggering while firing at the AI's board

## To test
1. Clone this repo and open in IntelliJ
2. Run all tests
3. Run Main and go to localhost:4567. Try placing ships such that their ending squares are offboard or their ending squares overlap with another ship; popups should be triggered
4. Verify that all four ships for the user and all four ships for the AI have visible lengths 5, 4, 3, 2, 2
5. Firing at AI's board should never trigger overlap/offboard error popups at any time
